### PR TITLE
coprocessor: add benchmarks about Json binary codec.

### PR DESCRIPTION
**Pull request by hicqu**
According to the benchmark result, we get conclusions:
1. for decode, there's no benefit between two formats;
2. for encode, binary format is much slower than text;

Why does the binary format performance worse? I think the reason
is we implement `JsonDecoder` for `Read` but not `Read + Seek`,
and implement `JsonEncoder` for `Write` but not `Write + Seek`.
It causes unneccessary copy of raw bytes in codec. For large and
complex Json, The influence is particularly significant.

I will raise another PR to fix this.

### CODE REVIEW

**Comment by hicqu** [comment]
Run new benchmarks:
```bash
cargo bench coprocessor::codec::mysql::json::binary::test -- --nocapture
```

@BusyJay @AndreMouche , PTAL, thanks.

File: src/coprocessor/codec/mysql/json/mod.rs [hunk 1]
```diff
@@ -151,4 +155,25 @@ mod test {
...
+        File::open(filename)
+            .map(BufReader::new)
+            .and_then(|reader| {
+                reader.lines().collect::<::std::io::Result<Vec<_>>>()
+            })
+            .unwrap()
+    }
+
+    pub fn load_test_jsons() -> Vec<String> {
+        let base = Path::new("src/coprocessor/codec/mysql/json");
```
Comment (BusyJay) [inline src/coprocessor/codec/mysql/json/mod.rs]: Use `CARGO_MANIFEST_DIR` instead just like https://github.com/pingcap/tikv/blob/master/tests/config/mod.rs#L40.

File: src/coprocessor/codec/mysql/json/mod.rs [hunk 2]
```diff
@@ -151,4 +155,26 @@ mod test {
...
+            .map(BufReader::new)
+            .and_then(|reader| {
+                reader.lines().collect::<::std::io::Result<Vec<_>>>()
+            })
+            .unwrap()
+    }
+
+    pub fn load_test_jsons() -> Vec<String> {
+        let mut file = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        file.push("src/coprocessor/codec/mysql/json/world_bank.json");
```
Comment (AndreMouche) [inline src/coprocessor/codec/mysql/json/mod.rs]: Is it weird to put the test data in the source directory?

**Comment by hicqu** [comment]
@AndreMouche , PTAL. Thanks.

**Comment by hicqu** [comment]
@AndreMouche , the new benchmarks are moved into `benches` now. PTAL, thanks.

**Pull request merged**